src/agent/src/util.rs

```diff
@@ -52,6 +57,27 @@ where
     Ok(total_bytes)
 }
 
+pub fn set_fd_close_exec(fd: RawFd) -> Result<RawFd> {
+    if let Err(e) = fcntl::fcntl(fd, FcntlArg::F_SETFD(FdFlag::FD_CLOEXEC)) {
+        return Err(anyhow!("failed to set fd: {} as close-on-exec: {}", fd, e));
+    }
+    Ok(fd)
+}
+
+pub fn get_vsock_incoming(fd: RawFd) -> Result<Incoming> {
+    set_fd_close_exec(fd)?;
```

Comment:
The listenFd had been set as CLOEXEC, why it needed to reset again here?

Reply:
Yes, it was a mistake, will remove it.

Reply:
done